BlueDB/src/main/java/org/bluedb/api/encryption/EncryptionServiceWrapper.java

```diff
@@ -0,0 +1,130 @@
+
+/**
+ * Wrapper class for interacting with an {@code EncryptionService} implementation optionally supplied on creation of a BlueDB instance.
+ */
+public class EncryptionServiceWrapper {
+
+	/**
+	 * An implementation of {@code EncryptionService} supplied on creation of a BlueDB instance, or null if none was provided.
+	 */
+	protected final EncryptionService encryptionService;
+
+	/**
+	 * A cached encryption version key populated when an instance of this class is created.
+	 * Used to detect changes to the wrapped {@code EncryptionService}, skipping unneeded validation of this field.
+	 */
+	protected String cachedEncryptionVersionKey;
+
+	public EncryptionServiceWrapper(EncryptionService encryptionService) {
+		this.encryptionService = encryptionService;
+		if (this.encryptionService != null) {
+			this.cachedEncryptionVersionKey = this.getCurrentEncryptionVersionKey();
+		}
+	}
+
+	/**
+	 * Encrypts and returns the bytes if encryption is enabled. Uses {@link EncryptionServiceWrapper#getCurrentEncryptionVersionKey()} to determine which encryption version to use.
+	 *
+	 * @param bytes the bytes to encrypt.
+	 * @return the encrypted bytes or the unmodified bytes if encryption is not enabled.
+	 */
+	public byte[] encryptOrReturn(byte[] bytes) {
+		if (isEncryptionEnabled()) {
+			return encrypt(this.getCurrentEncryptionVersionKey(), bytes);
+		}
+		return bytes;
+	}
+
+	/**
+	 * Encrypts and returns the bytes using a given encryption version key.
+	 *
+	 * @param encryptionVersionKey the encryption version key to use for encryption.
+	 * @param bytes the bytes to encrypt.
+	 * @return the encrypted bytes or the unmodified bytes if encryption is not enabled.
+	 */
+	public byte[] encryptOrThrow(String encryptionVersionKey, byte[] bytes) {
+		if (this.encryptionService != null) {
+			return this.encrypt(encryptionVersionKey, bytes);
+		}
+		throw new IllegalStateException("Unable to encrypt, encryption service not supplied");
+	}
+
+	/**
+	 * Decrypts and returns the bytes if encryption is enabled.
+	 *
+	 * @param encryptionVersionKey the encryption version key to use.
+	 * @param bytes                the bytes to decrypt.
+	 * @return the decrypted bytes or the unmodified bytes if encryption is not enabled.
+	 */
+	public byte[] decryptOrReturn(String encryptionVersionKey, byte[] bytes) {
+		if (isEncryptionEnabled()) {
+			return decrypt(encryptionVersionKey, bytes);
+		}
+		return bytes;
+	}
+
+	/**
+	 * Decrypts and returns the bytes if encryption is enabled and the {@code BlueFileMetadata} supplied indicates the file is encrypted. Uses the encryption version key from the file extension to decrypt.
+	 *
+	 * @param fileMetadata the {@link BlueFileMetadata} retrieved from the file with information regarding the encryption status of the file.
+	 * @param bytes    the bytes to decrypt.
+	 * @return the decrypted bytes or the unmodified bytes.
+	 */
+	public byte[] decryptOrReturn(BlueFileMetadata fileMetadata, byte[] bytes) {
```

Comment:
I'm open to this change, I was trying to avoid getting the version key if the encryption service is null and I try to flatten if statements where possible, but I agree that the readability is better with the nested if's in this case. Great callout, I like this a lot!